src/factories/RabbitMQChannelFactory.ts

```diff
@@ -0,0 +1,191 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+// noinspection JSMethodCanBeStatic
+
+import {inject, injectable} from 'inversify';
+import {TYPES} from '../constants/types';
+import {AmqpConnectionManager, ChannelWrapper} from 'amqp-connection-manager';
+import {ConfirmChannel, ConsumeMessage} from 'amqplib';
+import {
+    CensusClient,
+    Death,
+    FacilityControl,
+    GainExperience,
+    MetagameEvent,
+    PS2Event,
+    Stream,
+    VehicleDestroy,
+} from 'ps2census';
+import {getLogger} from '../logger';
+import ApplicationException from '../exceptions/ApplicationException';
+import {QueueMessageHandlerInterface} from '../interfaces/QueueMessageHandlerInterface';
+import {Options} from 'amqplib/properties';
+import AdminQueueMessage from '../data/AdminAggregator/AdminQueueMessage';
+
+@injectable()
+export default class RabbitMQChannelFactory {
+    private static readonly logger = getLogger('RabbitMQChannelFactory');
+    private channel: ChannelWrapper;
+
+    constructor(
+        @inject(TYPES.rabbitMqConnection) private readonly rabbit: AmqpConnectionManager,
+        private readonly censusClient: CensusClient,
+    ) {}
+
+    // Above this, QueueAuthority creates these streams
+    public create(
+        exchange: string,
+        queueName: string,
+        queueOptions: Options.AssertQueue = {},
+        pattern: string | null = null,
+        handler: QueueMessageHandlerInterface<any> | null = null,
+    ): ChannelWrapper {
+        this.channel = this.rabbit.createChannel({
+            json: true,
+            // eslint-disable-next-line @typescript-eslint/no-misused-promises
+            setup: async (channel: ConfirmChannel) => {
+                pattern = pattern ?? '#'; // Default to all messages if a pattern isn't supplied
+                queueOptions = {durable: true, ...queueOptions};
+
+                await Promise.all([
+                    channel.checkExchange(exchange),
+                    channel.assertQueue(queueName, queueOptions),
+                ]);
+                await channel.bindQueue(queueName, exchange, pattern);
+
+                this.channel.on('connect', () => {
+                    RabbitMQChannelFactory.logger.info(`[${queueName}] connected!`);
+                });
+
+                // eslint-disable-next-line @typescript-eslint/no-misused-promises
+                this.channel.on('close', async () => {
```

Comment:
This code is redundant, the `amqp-coonnection-manager` takes care of this.

Reply:
16e47883341e4f9a6216d7a1c1dd6758b2f520b7